ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/macos_codesign/tasks/main.yml

```diff
@@ -0,0 +1,124 @@
+---
+##################
+# macOS Codesign #
+##################
+
+- name: Set Jenkins user password
+  user:
+    name: jenkins
+    password: "{{ macOS_Jenkins_Password }}"
+
+- name: Create Jenkins password file
+  file: path="/Users/jenkins/.password" state=touch
+
+- name: Copy password to file
+  copy: content="{{ macOS_Jenkins_Password }}" dest="/Users/jenkins/.password"
+
+- name: Get macOS version
+  shell: sw_vers -productVersion
+  register: macos_version
+
+- name: Check if Default keychain is correct
+  shell: security default-keychain | tr -d '[:space:]' | sed 's/"//g'
+  register: default_keychain
+  become_user: jenkins
+  ignore_errors: true
+
+- name: Create Keychain if it doesn't exist
+  shell: security create-keychain -p "{{ macOS_Jenkins_Password }}" "/Users/jenkins/Library/Keychains/login.keychain-db"
+  become_user: jenkins
+  when:
+    - not macos_version | regex_search("10.10")
+    - default_keychain.stderr | regex_search("A default keychain could not be found")
+
+- name: Set Default keychain
+  shell: security default-keychain -s "/Users/jenkins/Library/Keychains/login.keychain-db"
+  become_user: jenkins
+  when:
+    - not macos_version | regex_search("10.10")
+    - default_keychain.stdout != "/Users/jenkins/Library/Keychains/login.keychain-db"
+
+- name: Reboot macOS after setting Default keychain
+  reboot:
+    reboot_timeout: 3600
+  when:
+    - not macos_version | regex_search("10.10")
+    - default_keychain.stdout != "/Users/jenkins/Library/Keychains/login.keychain-db"
+
+- name: Test Application Certificate
+  shell: |
+    security unlock-keychain -p `cat ~/.password` login.keychain-db
+    rm -rf /Users/jenkins/test && touch /Users/jenkins/test
+    codesign --sign "Developer ID Application: London Jamocha Community CIC" /Users/jenkins/test
```

Comment:
I wonder if we need to parameterize this for folks outside Adopt build farm using these scripts

Reply:
We have the AdoptOpenJDK tag here so no need.